migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+    permit_id                    integer                                                      ,
+    due_date                     timestamp                                NOT NULL            ,
+    submission_year              integer                                                      ,
+    deleted_ind                  boolean DEFAULT false                    NOT NULL            ,
+    create_user                  character varying(60)                    NOT NULL            ,
+    create_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    update_user                  character varying(60)                    NOT NULL            ,
+    update_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    
+    FOREIGN KEY (mine_report_definition_id) REFERENCES mine_report_definition(mine_report_definition_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_guid) REFERENCES mine(mine_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (permit_id) REFERENCES permit(permit_id) DEFERRABLE INITIALLY DEFERRED
+
+);
+
+COMMENT ON TABLE mine_report is 'A code required report for a specific mine or permit on a mine.';
+ALTER TABLE mine_report OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_submission (
+    mine_report_submission_id             serial                                  NOT NULL PRIMARY KEY,
+    mine_report_submission_guid           uuid                                    NOT NULL            ,
+    mine_report_id                        integer                                 NOT NULL            ,
+    received_date                         timestamp                                                   ,
+    mine_report_submission_status_code    varchar(3)                              NOT NULL            ,
+    active_ind                            boolean DEFAULT true                    NOT NULL            ,
+    create_user                           character varying(60)                   NOT NULL            ,
+    create_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+    update_user                           character varying(60)                   NOT NULL            ,
+    update_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+
+    FOREIGN KEY (mine_report_id) REFERENCES mine_report(mine_report_id) DEFERRABLE INITIALLY DEFERRED ,
+    FOREIGN KEY (mine_report_submission_status_code) REFERENCES mine_report_submission_status_code(mine_report_submission_status_code) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_submission is 'A file submission for a code required report.';
+ALTER TABLE mine_report_submission OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_document_xref
+(
+    mine_report_document_xref_guid uuid    DEFAULT gen_random_uuid() NOT NULL PRIMARY KEY,
```

Comment:
We may not need a surrogate primary key here... the combination of the mine_document_guid + mine_report_submission_id should be enough to keep the record unique and can be used as the primary key.

Reply:
Jason S and I had talked about this briefly and looking at our previous xref tables they all seem to use a separate primary key. I am all for using the combo as the PK but do we want to keep it consistent?

Reply:
I took a quick look at the other xref tables and there is a bit of a mix... I am all for consistency and probably missed the boat with the other xref tables, although some of them are good candidates for a surrogate key (and probably not "true" xref tables). I'm happy with whatever you guys decide.

Reply:
I dislike compounded keys. Highly prefer surrogates. Willing to talk offline about it.